src/app/ui/toc/symbology-stack.directive.js

```diff
@@ -120,16 +120,59 @@ function rvSymbologyStack($rootScope, $q, Geo, animationService, layerRegistry,
             // do nothing
         }
 
+        //TODO: find a better place for this function
+        // current plan is to get a list of valid OIDs for multiple query types (eg: filter by extent etc)
+        // enhancedTable will have a complete list of OIDs to show as calculated with multiple overlapping filters
+        function getSymbFilterOIDS(queryOpts) {
+            let legEntry = self.block;
+            legEntry.validOIDs = undefined;
+            const layerRecId = layerRegistry.getLayerRecord(legEntry.layerRecordId);
+            // query the layer itself instead of making a mapserver request
+            if (legEntry.parentLayerType === 'esriFeature' && legEntry.layerType === 'esriFeature' && layerRecId.dataSource() !== 'esri') {
+                queryOpts.featureLayer = layerRecId._layer;         // file based layer
+            } else {
+                queryOpts.url = legEntry.proxyWrapper.queryUrl;        // server based layer
+            }
+            gapiService.gapi.query.queryGeometry(queryOpts).then(featureSet => {
+                // save an array of OID's returned by the query
+                const validOIDs = featureSet.features.map(feat => parseInt(feat.attributes['OBJECTID']));
+                // set a list of validOIDs and fire an observable for the enhancedTable
+                legEntry.validOIDs = validOIDs;
+                if (legEntry.validOIDs !== undefined) {
+                    legEntry._symbolVisibilityChanged.next(true);
+                } else if (legEntry.validOIDs === undefined) {
+                    const proxyLoaded = $rootScope.$watch(() => legEntry.proxyWrapper.state, (state, oldState) => {
+                        if (state === 'rv-loaded') {
+                            legEntry._symbolVisibilityChanged.next(undefined);
```

Comment:
You can omit the `undefined`, otherwise you've defined an undefined 👍

Reply:
donethanks